status/condition_set_test.go

```diff
@@ -11,7 +11,7 @@ import (
 
 var _ = Describe("Conditions", func() {
 	It("should correctly toggle conditions", func() {
-		testObject := TestObject{}
+		testObject := &TestObject{ObjectMeta: metav1.ObjectMeta{Generation: 1}}
```

Comment:
Can we validate that an update that occurs that doesn't change the transition time still updates the observed generation on the condition type?

Reply:
Unit test also updated to cover such case